src/isomorphic/classic/element/ReactElementValidator.js

```diff
@@ -191,13 +191,30 @@ var ReactElementValidator = {
     // We warn in this case but don't throw. We expect the element creation to
     // succeed and there will likely be errors in render.
     if (!validType) {
-      warning(
-        false,
-        'React.createElement: type should not be null, undefined, boolean, or ' +
-          'number. It should be a string (for DOM elements) or a ReactClass ' +
-          '(for composite components).%s',
-        getDeclarationErrorAddendum()
-      );
+      if (
+        typeof type !== 'function' &&
+        typeof type !== 'string'
+      ) {
+        var info = '';
+        if (
+          typeof type === 'object' &&
+          type !== null &&
+          Object.keys(type).length === 0
+        ) {
+          info +=
+            ' You likely forgot to export your component from the file ' +
+            'it\'s defined in.';
+        }
+        info += getDeclarationErrorAddendum();
+        warning(
+          false,
+          'React.createElement: type is invalid -- expected a string (for ' +
+          'built-in components) or a class/function (for composite ' +
```

Comment:
Custom components makes me think of web components (custom web elements). I think user-defined sounds better 👍